third_party/ngraph/ngraph_tf.BUILD

```diff
@@ -0,0 +1,96 @@
+licenses(["notice"])  # 3-Clause BSD
+
+exports_files(["license.txt"])
+
+filegroup(
+    name = "LICENSE",
+    srcs = [
+        "license.txt",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+load(
+    "@org_tensorflow//tensorflow:tensorflow.bzl",
+    "tf_cc_test"
+)
+
+cc_library(
+    name = "ngraph_libs_linux",
+    srcs = [
+        "lib/libiomp5.so",
+        "lib/libmklml_intel.so",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
```

Comment:
Thanks for the fixes. There is one more request from our build team: could you make this a `cc_binary` with `linkshared=1` and test if it works? This aligns with TF's roadmap about tf.contrib and will help to make future integration/migration easier.

Other than that, this PR looks good to me.

Reply:
Thanks @aaroey for the feedback. I am going to try this an report back.

Reply:
@aaroey I was able to use `linkshared=1` a shared library for the unit tests (i.e., `ngraph_tf_tests`) but for the TF python module (i.e., `pywrap_tensorflow_internal`) I don't know how to link it. If you can provide some help that I can try that as well.

Reply:
@avijit-nervana Do you mean there were errors when linking it with `pywrap_tensorflow_internal`, or it can dynamically linked but not be able to be loaded when user `import tensorflow`?

Reply:
@aaroey I am seeing `bazel` errors. I was able to add the `@ngraph_tf//:libngraph_tf.so` to the `//tensorflow:libtensorflow_cc.so` target by adding the following line:
```
srcs = if_ngraph(["@ngraph_tf//:libngraph_tf.so"]),
```
However, don't know how to add this dependency to the `pywrap_tensorflow_internal` as this is a `tf_py_wrap_cc(...)` and looks like only soiurce it takes is a swig input file.

What's the recommended way to add the .so to the python wrapper? 

Also - if an external application uses the TensorFlow C++ API (e.g., a TensorFlow serving container) - then does the external application also need to link with `@ngraph_tf//:libngraph_tf.so`?

Reply:
So what we had in mind is, rather than explicit dynamic linking in tensorflow, ngraph build and users use this mechanism.

1 - When building, core TF only needs to expose the registration mechanism for optimizers.
2 - We build TF without ngraph
3 - You build ngraph to your specifications.
4 - Users load TF.
5 - Users dlopen ngraph library. This triggers a static initializer that registers ngraph optimizer with the core TF optimizer registry.
6 - Users then use ngraph code.

So following the above model:
- TF should not explicitly link against ngraph
- ngraph should explicitly link against libtensorflow_framework.so
- no explicit bazel dependency should be needed
- external applications can choose to explicitly link against ngraph, or they can simply dlopen ngraph. Either should just work.

Reply:
Thanks @avijit-nervana for the explanation and @gunan for the suggestions. I also tested the approach about `cc_binary` with `linkshared=1`, and it turns out it's not working since `cc_binary` is not expected in `tf_py_wrap_cc`, and we'll need to implement what @gunan says when the mechanism is ready.

At this moment, since the mechanism mentioned is not fully implemented/tested yet, I think we can keep the current approach and do the migration later which will make things easier and less error-prone.

Reply:
@aaroey Thanks for the approval. Since there is now a conflict with master - do you want me to resolve this?

Reply:
@avijit-nervana yes, please help to resolve it.

Reply:
@aaroey Done.